### ISSUE TITLE
Only playing segments

### ISSUE THREAD

**Issue by craigharvi3** [bbc/peaks.js#191]
Hey guys,

Great work on this. It works a treat.

Just wondering if it's possible, using the API - to only play segments when playing the audio via the audio controls?

For example:

<img width="972" alt="screen shot 2017-06-14 at 09 18 40" src="https://user-images.githubusercontent.com/22886093/27122403-80afef0a-50e2-11e7-933f-cf683fe3e032.png">

If I played the audio, the audio would start playing from the segment start and end playing at the segment end?

Cheers

Craig

**Comment by craigharvi3** [comment]
Not to worry. I found a solution using the API in the end.

**Comment by chrisn** [comment]
Thanks Craig. This is something we could consider adding to the library, maybe something along the lines of:
```javascript
peaks.player.playSegment(segment);
```

**Comment by craigharvi3** [comment]
Yup @chrisn - that'd be ideal.

Right now i'm controlling it with timers - so real dirty.

**Comment by chrisn** [comment]
I'd be happy to review a pull request based on monitoring the media element's `currentTime` in [player.js](https://github.com/bbc/peaks.js/blob/master/src/main/player/player.js).

**Comment by craigharvi3** [comment]
Sure @chrisn - hang tight.

**Comment by chrisn** [comment]
Resolved via #192.

**Comment by craigharvi3** [comment]
Hey @chrisn 
Any idea when this will be in a release?
Cheers.

**Comment by chrisn** [comment]
I think i'd aim to publish a new release by the end of the week. I have a couple of other changes to finish before it's ready.

**Comment by chrisn** [comment]
@craigharvi3 I've just published v0.8.0. Please let me know if anything doesn't work as expected.

**Comment by craigharvi3** [comment]
Great, will do. Thanks!

**Comment by craigharvi3** [comment]
Hey @chrisn

I'm seeing this error on the new version:

```
TypeError: self.deprecationLogger is not a function
    at Object.setCurrentTime (peaks.js:20655)
    at Peaks.<anonymous> (mix.js:57)
    at Peaks.EventEmitter.emit (peaks.js:383)
    at WaveformSegments.module.exports.WaveformSegments.init (peaks.js:21094)
    at Waveform.module.exports.Waveform.openZoomView (peaks.js:22060)
    at Peaks.<anonymous> (peaks.js:20587)
    at Peaks.EventEmitter.emit (peaks.js:386)
    at Waveform.module.exports.Waveform.handleRemoteData (peaks.js:22043)
    at onAudioDecoded (peaks.js:19397)
```

**Comment by craigharvi3** [comment]
Here's another error I'm seeing when calling `peaks.player.playSegment(id);`:

```
mix.js:23612 Uncaught TypeError: Failed to set the 'currentTime' property on 'HTMLMediaElement': The provided double value is non-finite.
    at Player.module.exports.Player.seek (mix.js:23612)
    at Player.module.exports.Player.playSegment (mix.js:23618)
    at Mix.trackSegment (mix.js:103)
    at HTMLAudioElement.<anonymous> (mix.js:95)
module.exports.Player.seek	@	mix.js:23612
module.exports.Player.playSegment	@	mix.js:23618
trackSegment	@	mix.js:103
(anonymous)	@	mix.js:95

```

**Comment by chrisn** [comment]
Apologies, I've just published a 0.8.1 bugfix release. Note that the `time.setCurrentTime()` is deprecated in favour of `player.seek()`.

`playSegment` accepts a segment object as a parameter, not a segment ID, so if you need to seek by ID you'll need to call `peaks.segments.getSegment(segmentId)`.

**Comment by craigharvi3** [comment]
Great stuff @chrisn 

Works a treat now. Thanks.
